### PR TITLE
Add gnuplot recipe

### PR DESCRIPTION
**Pull request by shahin**
Adapted from bioconda's recipe:

https://github.com/bioconda/bioconda-recipes/tree/0d6704d7165285acee0938608a513a5f06442997/recipes/gnuplot

### CODE REVIEW

File: recipes/gnuplot/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  license: Gnuplot License
+  license_file: Copyright
+  summary: Gnuplot, plotting from command line
+
+test:
+  files:
+    - test-data.txt
+  commands:
+    - gnuplot -e "set terminal dumb; set style histogram; p 'test-data.txt'"
+    - gnuplot -e "set terminal png"
```
Comment (patricksnape) [inline recipes/gnuplot/meta.yaml]: Please add yourself as a maintainer (in the `extras` section).

File: recipes/gnuplot/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    - libgd
+    - ncurses
+
+about:
+  home: https://github.com/gnuplot/gnuplot
+  license: Gnuplot License
+  license_file: Copyright
+  summary: Gnuplot, plotting from command line
+
+test:
```
Comment (patricksnape) [inline recipes/gnuplot/meta.yaml]: Please place the `test` section before the `about`.

File: recipes/gnuplot/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+package:
+  name: {{ pkgname }}
+  version: {{ version }}
+
+build:
+  number: 0
+  skip: True  # [win]
+
+source:
```
Comment (patricksnape) [inline recipes/gnuplot/meta.yaml]: Please place the `source` section before the `build`.

File: recipes/gnuplot/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  url: https://sourceforge.net/projects/{{ pkgname }}/files/{{ pkgname }}/{{ version }}/{{ pkgname }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+requirements:
+  build:
+    - gcc  # [not osx]
+    - llvm  # [osx]
+    - pkg-config  # [osx]
+    - libgd
+    - ncurses
```
Comment (patricksnape) [inline recipes/gnuplot/meta.yaml]: Looks like we may need iconv here? So that configure can find it?

**Comment by patricksnape** [comment]
Error on unix appears to be `/opt/rh/devtoolset-2/root/usr/bin/ld: set.o: undefined reference to symbol 'libiconv_open'`.

File: recipes/gnuplot/build.sh [hunk 5]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+./configure \
+    --prefix=$PREFIX \
+    --without-x \
+    --without-lua
```
Comment (patricksnape) [inline recipes/gnuplot/build.sh]: Complaining cairo isn't found, so we may want `--without-cairo` or to include cairo as a dep.

Comment (patricksnape) [inline recipes/gnuplot/build.sh]: We have our own `readline` build, so we may want to use `--with-readline=` and add `readline` as a dep.

Comment (patricksnape) [inline recipes/gnuplot/build.sh]: We don't have a latex stack so I would add `--without-latex`.

Comment (patricksnape) [inline recipes/gnuplot/build.sh]: We don't have `libcerf`, so I would add `--without-libcerf`

Comment (patricksnape) [inline recipes/gnuplot/build.sh]: We don't have `wx`, though I believe `wxpython` does exist on `defaults`. Not sure if we need the `--without-wx` flag or not.

File: recipes/gnuplot/build.sh [hunk 6]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+opts="
+    --without-x \
+    --without-lua \
+    --without-latex \
+    --without-libcerf \
+    --with-qt4
```
Comment (patricksnape) [inline recipes/gnuplot/build.sh]: Change to `--with-qt`

File: recipes/gnuplot/build.sh [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+    "
+
+opts="$opts --with-readline=$PREFIX"
```
Comment (patricksnape) [inline recipes/gnuplot/build.sh]: You can just merge this with the above

File: recipes/gnuplot/build.sh [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+
+LIBS="-liconv" ./configure --prefix=$PREFIX $opts
```
Comment (patricksnape) [inline recipes/gnuplot/build.sh]: Change to `LDFLAGS="$LDFLAGS -Wl,-rpath,$PREFIX/lib" LIBS="-liconv" ./configure --prefix=$PREFIX $opts`

File: recipes/gnuplot/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+    - cairo
+    - readline
+    - qt
+  run:
+    - libgcc  # [not osx]
```
Comment (patricksnape) [inline recipes/gnuplot/meta.yaml]: @conda-forge/core What exactly is the procedure with respect to `libgcc` etc? I can't keep track 😆

File: recipes/gnuplot/Copyright [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+ *    support of your modified version, and
+ *   4. retain our contact information in regard to use of the base
+ *    software.
+ * Permission to distribute the released version of the source code along
+ * with corresponding source modifications in the form of a patch file is
+ * granted with same provisions 2 through 4 for binary distributions.
+ *
+ * This software is provided "as is" without express or implied warranty
+ * to the extent permitted by applicable law.
+]*/
```
Comment (jakirkham) [inline recipes/gnuplot/Copyright]: Should we drop the comment? Seems like this is in a C-style comment.

**Comment by patricksnape** [comment]
Made a couple of comments - can confirm this fixes the `iconv` linking errors on my Mac.

File: recipes/gnuplot/meta.yaml (hunk 9 above)
Comment (croth1) [inline recipes/gnuplot/meta.yaml]: The gcc/libgcc/llvm dependencies here are specific to bioconda's conventions. conda-forge seems to use the native compiler, so these dependencies should be removed. Also note that some of the dependencies need pinning: See https://github.com/conda-forge/staged-recipes/wiki/Pinned-dependencies

**Comment by shahin** [comment]
I noticed the `_placehold_placehold_placeho...` that shows up in `PREFIX` on mac/circleci when I upgraded conda-build from 1.21.5 to 1.21.11. Downgrading back to 1.21.5 fixed it locally. Anybody know what the issue is?

**Comment by jakirkham** [comment]
No clue. Though it seems to not be present in `1.21.9` either. Raised an issue ( https://github.com/conda/conda-build/issues/1184 ) upstream to get more details.

**Comment by patricksnape** [comment]
I don't think that placeholder matters as long as it's consistent.

File: recipes/gnuplot/meta.yaml (hunk 1 above)
Comment (jakirkham) [inline recipes/gnuplot/meta.yaml]: Done.

File: recipes/gnuplot/meta.yaml [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+test:
+  files:
+    - test-data.txt
+  commands:
+    - gnuplot -e "set terminal dumb; set style histogram; p 'test-data.txt'"
+    - gnuplot -e "set terminal png"
+
+about:
+  home: https://github.com/gnuplot/gnuplot
+  license: Gnuplot License
```
Comment (jakirkham) [inline recipes/gnuplot/meta.yaml]: Apparently some files are dual-licensed. Would be great to figure out what those other licenses are and add them here. Often Linux distros do a pretty good job of this stuff. So checking a few Linux distros may give us a good idea of what is going on.

File: recipes/gnuplot/meta.yaml [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+    - readline ==6.2
+    - libgd
+    - cairo
+    - qt
+  run:
+    - ncurses ==5.9
+    - readline ==6.2
+    - libgd
+    - cairo
+    - qt
```
Comment (jakirkham) [inline recipes/gnuplot/meta.yaml]: Probably going to want to pin `qt` somehow.

File: recipes/gnuplot/meta.yaml [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+
+requirements:
+  build:
+    - ncurses ==5.9
+    - readline ==6.2
+    - libgd
+    - cairo
+    - qt
+  run:
+    - ncurses ==5.9
```
Comment (jakirkham) [inline recipes/gnuplot/meta.yaml]: Please change to `ncurses 5.9*`.

File: recipes/gnuplot/meta.yaml [hunk 14]
```diff
@@ -0,0 +1,47 @@
...
+  url: https://sourceforge.net/projects/{{ pkgname }}/files/{{ pkgname }}/{{ version }}/{{ pkgname }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  skip: True  # [win]
+
+requirements:
+  build:
+    - ncurses ==5.9
```
Comment (jakirkham) [inline recipes/gnuplot/meta.yaml]: Please change to `ncurses 5.9*`.

File: recipes/gnuplot/meta.yaml [hunk 15]
```diff
@@ -0,0 +1,47 @@
...
+    - readline ==6.2
```
Comment (jakirkham) [inline recipes/gnuplot/meta.yaml]: Please change to `readline 6.2*`.

File: recipes/gnuplot/meta.yaml [hunk 16]
```diff
@@ -0,0 +1,47 @@
...
+    - libgd
+    - cairo
+    - qt
+  run:
+    - ncurses ==5.9
+    - readline ==6.2
```
Comment (jakirkham) [inline recipes/gnuplot/meta.yaml]: Please change to `readline 6.2*`.

**Comment by jakirkham** [comment]
Please add this [`yum_requirements.txt`]( https://github.com/conda-forge/qt-feedstock/blob/cb425417c8303c4e4d1b52099c34c76957b178b6/recipe/yum_requirements.txt ) to your recipe.

**Comment by jakirkham** [comment]
Thoughts on this @shahin ?

**Comment by jakirkham** [comment]
Closing as this seems to be resolved with the merge of PR ( https://github.com/conda-forge/staged-recipes/pull/2066 ) built largely off of the nice contribution here. Thanks for getting this started @shahin .

Feel free to add yourself on the [feedstock]( https://github.com/conda-forge/gnuplot-feedstock ) if you are still interested in keeping this up.

**Pull request closed without merging**